testament/tests/shouldfail/tccodecheck.nim

```diff
@@ -1,5 +1,4 @@
 discard """
-  targets: "c"
```

Comment:
reverts workarounds added in https://github.com/nim-lang/Nim/pull/16698, refs https://github.com/nim-lang/Nim/pull/16698#pullrequestreview-627522307, thanks to the new way tests/testament/tshould_not_work.nim is written